_episodes/18.md

```diff
@@ -0,0 +1,560 @@
+
+Automatically keeping materialized views fresh can be tricky from resource 
+management point of view since the computation to materialize the materialized 
+view can be expensive. Trino currently does not support automatic refresh of 
+materialized views. It instead supports the `REFRESH MATERIALIZED VIEW` command 
+that the user can issue to ensure that the materialized view is fresh.
+
+As a part of executing `REFRESH MATERIALIZED VIEW` command in Trino, existing
+data in the materialized view drops and new/fresh data re-inserted if there have been any 
+changes to base data. If the base data has not changed at all, the 
+`REFRESH MATERIALIZED VIEW` command is a no-op. 
+
+What happens if the user issues a query against the materialized view, and the 
+materialized view is not fresh? Trino detects that the materialized view is 
+stale, so it expands the materialized view definition, much like a logical view 
+and executes that SQL statement. Trino runs the query against the base tables.
+
+Incremental or delta refresh of materialized views is a more efficient way of
+keeping the materialized view in sync with the base data. An incremental refresh 
+means only parts of the data that need to be updated in MV are updated, rest of 
+the data is left untouched. (An example: base table `sales` partitioned on 
+`date` column, where the `sales` table gets data for that day inserted. If the 
+materialized view is also partitioned on `date`, a new partition for a day can 
+be added and data inserted for that day. Data for previous days/months is still 
+fresh and can be left untouched). This is something on our roadmap. Some of the 
+ways of incrementally refresh the materialized view can be partition level 
+refresh, another can be a more granular row-level refresh by using 
+functionality similar to `SQL MERGE` statement.
+
+
+### Support in Trino and at Netflix:
+
+#### Netflix materialized views
+
+The main reason Netflix is interested in materialized views is to give analysts 
+an easy way to compute and materialize their frequently used queries and keep 
+the results refreshed without relying on ETL pipeline to create and maintain 
+those result sets. Some materialized views are as simple as queries that project
+columns and apply filters, selecting data for a time range or for a test-id, 
+others are more complex that perform multi-level joins and aggregations.
+
+#### Netflix materialized view cross compatibility extension 
+
+Materialized views, much like logical views are compatible across the two main 
+engines used at Netflix, Trino and Spark. A little more about how Netflix uses 
+Trino to understand why logical views and materialized views are important. 
+Spark is used at Netflix to do ETL, creating and populating tables, while Trino 
+is the most popular engine with analysts and developers for adhoc/experimental 
+queries as well as audits. 
+
+Trino is also used for CTAS in some use cases. Both the engines access data from
```

Comment:
is also use to insert data with CREATE TABLE AS SELECT (CTAS) statements in some use cases